### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by kinghua0629**
This pull request introduces several new extracurricular activities to the in-memory database, enhances the frontend to display participants for each activity, and adds robust backend and frontend validation for activity signups. It also introduces a comprehensive test suite to ensure API correctness and data integrity.

**Backend improvements:**

* Added six new activities (`Basketball Team`, `Soccer Club`, `Art Club`, `Drama Club`, `Debate Team`, `Math Club`) to the `activities` dictionary in `app.py`, each with descriptions, schedules, and participant tracking.
* Updated the `signup_for_activity` function to prevent duplicate signups by checking if a student's email is already in the participants list, raising a 400 error if so.

**Frontend enhancements:**

* Modified the frontend JavaScript (`app.js`) to display a formatted list of participants for each activity, or a placeholder if there are none. The activities list now refreshes after a successful signup to reflect changes. [[1]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R23-R38) [[2]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R76-R78)
* Improved UI styling for participant lists and sections in `styles.css` for better readability and user experience.
* Added a sample participants list and unregister button (with placeholder logic) to `index.html` for future participant management features.

**Testing:**

* Added a new test suite in `tests/test_app.py` covering activity retrieval, successful and duplicate signups, participant count accuracy, and edge cases such as special character emails.
* Initialized the test package with `tests/__init__.py`.

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -62,6 +98,10 @@ def signup_for_activity(activity_name: str, email: str):
     # Get the specific activity
     activity = activities[activity_name]
 
+    # Validate student is not already signed up
+    if email in activity["participants"]:
+        raise HTTPException(status_code=400, detail="Student already signed up for this activity")
+
     # Add student
     activity["participants"].append(email)
```
[the remark this answers is not in the document]
Comment (kinghua0629) [reply src/app.py:106]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/kinghua0629/skills-getting-started-with-github-copilot/pull/2#discussion_r2658855652)

File: tests/test_app.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        "Programming Class",
+        "Gym Class"
+    ]
+    
+    response = client.get("/activities")
+    data = response.json()
+    
+    for activity in expected_activities:
+        assert activity in data
+        assert data[activity]["max_participants"] > 0
```
[the remark this answers is not in the document]
Comment (kinghua0629) [reply tests/test_app.py:187]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/kinghua0629/skills-getting-started-with-github-copilot/pull/2#discussion_r2658855659)

File: src/app.py [hunk 3]
```diff
@@ -53,7 +89,7 @@ def get_activities():
 
 
 @app.post("/activities/{activity_name}/signup")
-def signup_for_activity(activity_name: str, email: str):
+def signup_for_activity(activity_name: str, email: str):
```
[the remark this answers is not in the document]
Comment (kinghua0629) [reply src/app.py:92]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/kinghua0629/skills-getting-started-with-github-copilot/pull/2#discussion_r2658855660)

File: src/static/app.js [hunk 4]
```diff
@@ -20,11 +20,22 @@ document.addEventListener("DOMContentLoaded", () => {
 
         const spotsLeft = details.max_participants - details.participants.length;
 
+        // Create participants list HTML
+        const participantsList = details.participants.length > 0
+          ? `<ul class="participants-list">
+              ${details.participants.map(email => `<li>${email}</li>`).join('')}
+            </ul>`
```
[the remark this answers is not in the document]
Comment (kinghua0629) [reply src/static/app.js:27]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/kinghua0629/skills-getting-started-with-github-copilot/pull/2#discussion_r2658855663)

File: src/static/app.js [hunk 5]
```diff
@@ -62,6 +73,9 @@ document.addEventListener("DOMContentLoaded", () => {
         messageDiv.textContent = result.message;
         messageDiv.className = "success";
         signupForm.reset();
+        
+        // Refresh activities list to show updated participants
+        fetchActivities();
```
[the remark this answers is not in the document]
Comment (kinghua0629) [reply src/static/app.js:78]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/kinghua0629/skills-getting-started-with-github-copilot/pull/2#discussion_r2658855672)

File: tests/test_app.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        ("test+tag@mergington.edu", "Math Club")
+    ]
+    
+    for email, activity in test_cases:
+        response = client.post(
+            f"/activities/{quote(activity)}/signup?email={quote(email)}"
+        )
+        assert response.status_code == 200
+        assert email in activities[activity]["participants"]
+
```
[the remark this answers is not in the document]
Comment (kinghua0629) [reply tests/test_app.py:166]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/kinghua0629/skills-getting-started-with-github-copilot/pull/2#discussion_r2658855677)

**Comment by kinghua0629** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/kinghua0629/skills-getting-started-with-github-copilot/pull/2#pullrequestreview-3624297125)
